lib/smart_proxy_openscap/arf_html.rb

```diff
@@ -1,21 +1,28 @@
-require 'openscap'
-require 'openscap/ds/arf'
+require 'smart_proxy_openscap/storage_fs'
+require 'smart_proxy_openscap/openscap_exception'
 
 module Proxy
   module OpenSCAP
     class ArfHtml
-      def generate_html(file_in, file_out)
-        ::OpenSCAP.oscap_init
-        File.write file_out, get_arf_html(file_in)
-      ensure
-        ::OpenSCAP.oscap_cleanup
+      include ::Proxy::Log
+
+      def generate(cname, id, date, digest)
+        file_path = file_path_in_storage cname, id, date, digest
+        as_html file_path
+      end
+
+      def as_html(file_in_storage)
+        `oscap xccdf generate report #{file_in_storage}`
+      rescue => e
```

Comment:
This can probably raise some FileNotFound exception if `oscap` isn't in `$PATH`.